ocaml/xapi/server_helpers.ml

```diff
@@ -37,17 +37,27 @@ let async_wire_name = "Async."
 
 let async_length = String.length async_wire_name
 
+let internal_async_wire_name = "InternalAsync."
+
+let internal_async_length = String.length internal_async_wire_name
+
 (* hardcode the wire-name messages that we want to supress the printing of in the logs to avoid log spam: *)
 let supress_printing_for_these_messages : (string,unit) Hashtbl.t =
   let tbl = Hashtbl.create 20 in
   List.iter (fun k -> Hashtbl.replace tbl k ()) ["host.tickle_heartbeat"; "session.login_with_password"; "session.logout"; "session.local_logout"; "session.slave_local_login"; "session.slave_local_login_with_password"];
   tbl
 
-let is_async x =
-  String.length x > async_length && (String.sub x 0 async_length = async_wire_name)
-
-let remove_async_prefix x =
-  String.sub x async_length (String.length x - async_length)
+(** removes Async. or InternalAsync. prefixes if they exist
+  * NB. X_y api call format does not work with Async (and didn't previously) *)
+let sync_ty_and_maybe_remove_prefix x =
+  let len_x = String.length x in
+  (* expect more async calls than internal_async, so check for async first *)
+  if len_x > async_length && Astring.String.is_prefix ~affix:async_wire_name x then
```

Comment:
Is it the first check needed for performance? (It's not needed functionally)

Reply:
no, i can get rid of it (it's a hangover from how it was done before - using `String.sub`)